vector/src/main/scala/geotrellis/vector/affine/package.scala

```diff
@@ -0,0 +1,56 @@
+package geotrellis.vector
+
+import com.vividsolutions.jts.{geom => jts}
+
+package object affine {
+
+  implicit class AffineTransformationPolygon(geom: Polygon) {
```

Comment:
This is right on track. However I'd break it out into a separate trait, so the two files would look like:

```scala
package geotrellis.vector

...

package object affine {
  implicit class PolygonTransformations(val geom: Polygon) extends PolygonTransformationMethods
}
```

```scala
package geotrellis.vector.affine

trait PolygonTransformationMethods extends Any {
  val geom: Polygon

  def reflect(x: Double, y: Double): Polygon = { ... }
  ...  
}
```

This allows for better code organization, as you create each of the implicit classes to add methods to the different geometry types